### ISSUE TITLE
5% of titles are "unavailable" but do appear in my liibrary and are playable - what to do?

### ISSUE THREAD

**Issue by thefearlessleader** [rmcrackan/Libation#1045]
**5% of my titles are "unavailable" but do appear in my library and are playable - what to do?**

Libation shows 5% of the titles as "unavailable". I've quit/relaunched and logged out and re-logged in and scanned, but they all keep showing as "unavailable".

I've also tried one specific title as a test case, doing:
`LibationCli set-status --force -n B078XL1VVM`\
and then\
`LibationCli liberate B078XL1VVM`
`DownloadDecryptBook Begin: 6/17/2024 [B078XL1VVM] No Time to Spare`
`DownloadDecryptBook Completed: 6/17/2024 [B078XL1VVM] No Time to Spare`
`Error processing book. Skipping. This book will be tried again on next attempt. For options of skipping or marking as error, retry with main Libation app.. See log for more details.`
`Done. All books have been processed`

Not sure it's a bug, but I'm happy to try different things to see if I can "do" something to get this to work...\
Basically: help, what can I try to get these 5% of titles "found" and "downloaded".

**Platform**

MacOS (13.7), Libation (11.1.0.1)

**Comment by rmcrackan** [comment]
Are any of your unavailable titles Plus Catalog titles which are no longer available?

**Comment by thefearlessleader** [comment]
At least the one test title is definitely available (I can look it up, it's "in your library", and is purchasable, and is playable using the app) and I did actually purchase it. I believe that's also true for all of the other "unavailable" titles -- or at least the vast majority of those titles.

**Comment by rmcrackan** [comment]
Please find a title which you can play in the official app then try to download it with Libation. When that fails, please attach the latest log file to this ticket. That way, the relevant failure should be easy to find near the end of the log and I can take a look at the details.

**Comment by thefearlessleader** [comment]
> Please find a title which you can play in the official app then try to download it with Libation. When that fails, please attach the latest log file to this ticket. That way, the relevant failure should be easy to find near the end of the log and I can take a look at the details.


I re-ran the cli that I noted above (for one specific title that I DO own, Audible shows it is in my library, I can play on the Audible app, Audible shows it is available for purchase, etc.). 

In the log I see two likely reasons for it not working, but no clue WHY this is happening, though;
> Content License denied for asin: [B078XL1VVM] 

and

> Asin: [B078XL1VVM] is not eligible for AYCL

So, for you... Here's the log of that cli attempt...

> {}2024-11-17 20:48:43.404 -08:00 [INF] (at Dinah.Core.IO.MultiTextWriter.WriteLine(System.String)) DownloadDecryptBook Begin: 6/17/2024 [B078XL1VVM] No Time to Spare
 {}2024-11-17 20:48:43.483 -08:00 [INF] (at AudibleUtilities.ApiExtended+<CreateAsync>d__9.MoveNext()) {"Username":"x[...]z@x[...]z.c[...]m","LocaleName":"us"}
 {}2024-11-17 20:48:56.539 -08:00 [ERR] (at AudibleApi.Api+<GetDownloadLicenseAsync>d__24.MoveNext()) Content License denied
AudibleApi.ContentLicenseDeniedException: Content License denied for asin: [B078XL1VVM]
 {"ExceptionDetail":{"HResult":-2146233088,"Message":"Content License denied for asin: [B078XL1VVM]","Source":null,"Client":{"Message":"urn:cdo:AudibleApiExternalRouterService:Prod:Default[AAA], does not has access to asin[B078XL1VVM].urn:cdo:AudibleApiExternalRouterService:Prod:Default[LWA], does not has access to asin[B078XL1VVM].","RejectionReason":"RequesterEligibility","ValidationType":"Client"},"Ownership":{"Message":"Ownership: No Ownership information returned by DAOQS for customer [##############] and for asin [B078XL1VVM].","RejectionReason":"RequesterEligibility","ValidationType":"Ownership"},"Membership":{"Message":"Customer is not part of any plans","RejectionReason":"RequesterEligibility","ValidationType":"Membership"},"AYCL":{"Message":"Asin: [B078XL1VVM] is not eligible for AYCL","RejectionReason":"ContentEligibility","ValidationType":"AYCL"},"RequestUri":"https://api.audible.com/1.0/content/B078XL1VVM/licenserequest","JsonMessage":"{\"license_denial_reasons\":[{\"message\":\"Customer is not part of any plans\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Membership\"},{\"message\":\"Ownership: No Ownership information returned by DAOQS for customer [##############] and for asin [B078XL1VVM].\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Ownership\"},{\"message\":\"urn:cdo:AudibleApiExternalRouterService:Prod:Default[AAA], does not has access to asin[B078XL1VVM].urn:cdo:AudibleApiExternalRouterService:Prod:Default[LWA], does not has access to asin[B078XL1VVM].\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Client\"},{\"message\":\"Asin: [B078XL1VVM] is not eligible for AYCL\",\"rejectionReason\":\"ContentEligibility\",\"validationType\":\"AYCL\"}]}","Type":"AudibleApi.ContentLicenseDeniedException"}}2024-11-17 20:48:56.555 -08:00 [INF] (at Dinah.Core.IO.MultiTextWriter.WriteLine(System.String)) DownloadDecryptBook Completed: 6/17/2024 [B078XL1VVM] No Time to Spare
 {}2024-11-17 20:48:56.567 -08:00 [ERR] (at LibationCli.ProcessableOptionsBase+<ProcessOneAsync>d__6.MoveNext()) Error processing book. Skipping. This book will be tried again on next attempt. For options of skipping or marking as error, retry with main Libation app. {"Book":["B078XL1VVM","No Time to Spare","us","x[...]z@x[...]z.c[...]m"]}
AudibleApi.ContentLicenseDeniedException: Content License denied for asin: [B078XL1VVM]
   at AudibleApi.Api.GetDownloadLicenseAsync(String asin, DownloadQuality quality, ChapterTitlesType chapterTitlesType) in /home/runner/work/AudibleApi/AudibleApi/AudibleApi/Api.Download.cs:line 155
   at FileLiberator.DownloadDecryptBook.downloadAudiobookAsync(LibraryBook libraryBook) in /Users/runner/work/Libation/Libation/Source/FileLiberator/DownloadDecryptBook.cs:line 127
   at FileLiberator.DownloadDecryptBook.ProcessAsync(LibraryBook libraryBook) in /Users/runner/work/Libation/Libation/Source/FileLiberator/DownloadDecryptBook.cs:line 56
   at FileLiberator.Processable.ProcessSingleAsync(LibraryBook libraryBook, Boolean validate) in /Users/runner/work/Libation/Libation/Source/FileLiberator/Processable.cs:line 54
   at LibationCli.ProcessableOptionsBase.ProcessOneAsync(Processable Processable, LibraryBook libraryBook, Boolean validate) in /Users/runner/work/Libation/Libation/Source/LibationCli/Options/_ProcessableOptionsBase.cs:line 78
 {"ExceptionDetail":{"HResult":-2146233088,"Message":"Content License denied for asin: [B078XL1VVM]","Source":"AudibleApi","TargetSite":"Void MoveNext()","Client":{"Message":"urn:cdo:AudibleApiExternalRouterService:Prod:Default[AAA], does not has access to asin[B078XL1VVM].urn:cdo:AudibleApiExternalRouterService:Prod:Default[LWA], does not has access to asin[B078XL1VVM].","RejectionReason":"RequesterEligibility","ValidationType":"Client"},"Ownership":{"Message":"Ownership: No Ownership information returned by DAOQS for customer [##############] and for asin [B078XL1VVM].","RejectionReason":"RequesterEligibility","ValidationType":"Ownership"},"Membership":{"Message":"Customer is not part of any plans","RejectionReason":"RequesterEligibility","ValidationType":"Membership"},"AYCL":{"Message":"Asin: [B078XL1VVM] is not eligible for AYCL","RejectionReason":"ContentEligibility","ValidationType":"AYCL"},"RequestUri":"https://api.audible.com/1.0/content/B078XL1VVM/licenserequest","JsonMessage":"{\"license_denial_reasons\":[{\"message\":\"Customer is not part of any plans\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Membership\"},{\"message\":\"Ownership: No Ownership information returned by DAOQS for customer [##############] and for asin [B078XL1VVM].\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Ownership\"},{\"message\":\"urn:cdo:AudibleApiExternalRouterService:Prod:Default[AAA], does not has access to asin[B078XL1VVM].urn:cdo:AudibleApiExternalRouterService:Prod:Default[LWA], does not has access to asin[B078XL1VVM].\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Client\"},{\"message\":\"Asin: [B078XL1VVM] is not eligible for AYCL\",\"rejectionReason\":\"ContentEligibility\",\"validationType\":\"AYCL\"}]}","Type":"AudibleApi.ContentLicenseDeniedException"}}2024-11-17 20:48:56.584 -08:00 [INF] (at Dinah.Core.IO.MultiTextWriter.WriteLine(System.String)) Done. All books have been processed

**Comment by alex-ries** [comment]
Adding on to this thread. I seem to have the exact same issue with just 2 titles. I get the same error as @thefearlessleader : 

` {}2024-12-02 08:26:25.890 -05:00 [INF] (at FileLiberator.Processable+<ProcessSingleAsync>d__23.MoveNext()) Begin ProcessSingleAsync {"TitleWithSubtitle":"Columbus Day: Expeditionary Force, Book 1","AudibleProductId":"B01N48VJFJ","Locale":"us","Account":"a[...]x@f[...]s.n[...]l"}
 {}2024-12-02 08:26:25.936 -05:00 [INF] (at Dinah.Core.IO.MultiTextWriter.WriteLine(System.String)) DownloadDecryptBook Begin: 12/11/2022 [B01N48VJFJ] Columbus Day: Expeditionary Force, Book 1
 {}2024-12-02 08:26:26.010 -05:00 [INF] (at AudibleUtilities.ApiExtended+<CreateAsync>d__9.MoveNext()) {"Username":"a[...]x@f[...]s.n[...]l","LocaleName":"us"}
 {}2024-12-02 08:26:27.145 -05:00 [ERR] (at AudibleApi.Api+<GetDownloadLicenseAsync>d__24.MoveNext()) Content License denied
AudibleApi.ContentLicenseDeniedException: Content License denied for asin: [B01N48VJFJ]
 {"ExceptionDetail":{"HResult":-2146233088,"Message":"Content License denied for asin: [B01N48VJFJ]","Source":null,"Client":{"Message":"urn:cdo:AudibleApiExternalRouterService:Prod:Default[AAA], does not has access to asin[B01N48VJFJ].urn:cdo:AudibleApiExternalRouterService:Prod:Default[LWA], does not has access to asin[B01N48VJFJ].","RejectionReason":"RequesterEligibility","ValidationType":"Client"},"Ownership":{"Message":"Ownership: No Ownership information returned by DAOQS for customer [##############] and for asin [B01N48VJFJ].","RejectionReason":"RequesterEligibility","ValidationType":"Ownership"},"Membership":{"Message":"Customer is not part of any plans","RejectionReason":"RequesterEligibility","ValidationType":"Membership"},"AYCL":{"Message":"Asin: [B01N48VJFJ] is not eligible for AYCL","RejectionReason":"ContentEligibility","ValidationType":"AYCL"},"RequestUri":"https://api.audible.com/1.0/content/B01N48VJFJ/licenserequest","JsonMessage":"{\"license_denial_reasons\":[{\"message\":\"Customer is not part of any plans\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Membership\"},{\"message\":\"Ownership: No Ownership information returned by DAOQS for customer [##############] and for asin [B01N48VJFJ].\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Ownership\"},{\"message\":\"urn:cdo:AudibleApiExternalRouterService:Prod:Default[AAA], does not has access to asin[B01N48VJFJ].urn:cdo:AudibleApiExternalRouterService:Prod:Default[LWA], does not has access to asin[B01N48VJFJ].\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Client\"},{\"message\":\"Asin: [B01N48VJFJ] is not eligible for AYCL\",\"rejectionReason\":\"ContentEligibility\",\"validationType\":\"AYCL\"}]}","Type":"AudibleApi.ContentLicenseDeniedException"}}2024-12-02 08:26:27.199 -05:00 [INF] (at Dinah.Core.IO.MultiTextWriter.WriteLine(System.String)) DownloadDecryptBook Completed: 12/11/2022 [B01N48VJFJ] Columbus Day: Expeditionary Force, Book 1
 {}2024-12-02 08:26:27.240 -05:00 [ERR] (at LibationCli.ProcessableOptionsBase+<ProcessOneAsync>d__6.MoveNext()) Error processing book. Skipping. This book will be tried again on next attempt. For options of skipping or marking as error, retry with main Libation app. {"Book":["B01N48VJFJ","Columbus Day: Expeditionary Force, Book 1","us","a[...]x@f[...]s.n[...]l"]}
AudibleApi.ContentLicenseDeniedException: Content License denied for asin: [B01N48VJFJ]
   at AudibleApi.Api.GetDownloadLicenseAsync(String asin, DownloadQuality quality, ChapterTitlesType chapterTitlesType) in /home/runner/work/AudibleApi/AudibleApi/AudibleApi/Api.Download.cs:line 155
   at FileLiberator.DownloadDecryptBook.downloadAudiobookAsync(LibraryBook libraryBook) in /Source/FileLiberator/DownloadDecryptBook.cs:line 127
   at FileLiberator.DownloadDecryptBook.ProcessAsync(LibraryBook libraryBook) in /Source/FileLiberator/DownloadDecryptBook.cs:line 56
   at FileLiberator.Processable.ProcessSingleAsync(LibraryBook libraryBook, Boolean validate) in /Source/FileLiberator/Processable.cs:line 54
   at LibationCli.ProcessableOptionsBase.ProcessOneAsync(Processable Processable, LibraryBook libraryBook, Boolean validate) in /Source/LibationCli/Options/_ProcessableOptionsBase.cs:line 78
 {"ExceptionDetail":{"HResult":-2146233088,"Message":"Content License denied for asin: [B01N48VJFJ]","Source":"AudibleApi","TargetSite":"Void MoveNext()","Client":{"Message":"urn:cdo:AudibleApiExternalRouterService:Prod:Default[AAA], does not has access to asin[B01N48VJFJ].urn:cdo:AudibleApiExternalRouterService:Prod:Default[LWA], does not has access to asin[B01N48VJFJ].","RejectionReason":"RequesterEligibility","ValidationType":"Client"},"Ownership":{"Message":"Ownership: No Ownership information returned by DAOQS for customer [##############] and for asin [B01N48VJFJ].","RejectionReason":"RequesterEligibility","ValidationType":"Ownership"},"Membership":{"Message":"Customer is not part of any plans","RejectionReason":"RequesterEligibility","ValidationType":"Membership"},"AYCL":{"Message":"Asin: [B01N48VJFJ] is not eligible for AYCL","RejectionReason":"ContentEligibility","ValidationType":"AYCL"},"RequestUri":"https://api.audible.com/1.0/content/B01N48VJFJ/licenserequest","JsonMessage":"{\"license_denial_reasons\":[{\"message\":\"Customer is not part of any plans\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Membership\"},{\"message\":\"Ownership: No Ownership information returned by DAOQS for customer [##############] and for asin [B01N48VJFJ].\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Ownership\"},{\"message\":\"urn:cdo:AudibleApiExternalRouterService:Prod:Default[AAA], does not has access to asin[B01N48VJFJ].urn:cdo:AudibleApiExternalRouterService:Prod:Default[LWA], does not has access to asin[B01N48VJFJ].\",\"rejectionReason\":\"RequesterEligibility\",\"validationType\":\"Client\"},{\"message\":\"Asin: [B01N48VJFJ] is not eligible for AYCL\",\"rejectionReason\":\"ContentEligibility\",\"validationType\":\"AYCL\"}]}","Type":"AudibleApi.ContentLicenseDeniedException"}}2024-12-02 08:26:27.286 -05:00 [INF] (at FileLiberator.Processable+<ProcessSingleAsync>d__23.MoveNext()) Begin ProcessSingleAsync {"TitleWithSubtitle":"Roseland, Volume 1","AudibleProductId":"1774243997","Locale":"us","Account":"a[...]x@f[...]s.n[...]l"}`

I run Libation headless in a docker container, and it runs great on most of my library it's just 2 specific titles that don't work (I only posted one, but the other looks exactly the same)

2 things I checked/noticed:
First:the ASIN seems correct vs what the audible website tells me, plus the page says "in library" neither of these two books were part of audible plus. I bought both of them.
Second: when using the Libation windows UI the UI tells me "This book cannot be dowloaded because it was not found in the most recent library scan" which is weird because I deleted it from my library, then deleted the deleted book through hangover, rescanned and it showed up in that state.

I have a pretty big library so like @thefearlessleader this is a fairly small percentage of the total books that _do_ work great.

**Comment by alex-ries** [comment]
Following up after doing some research, I think that one of the books was free for one reason or another when it was added to my library, and the other may have been removed. So I got it, it got added to my library, now it errors out.
My fix: I removed both from the library so it won't show up at all and won't error. I may get these books elsewhere.

Also something I tried was use audible-cli just to see what's under the hood and it gave the same "book is not downloadable" even though when I query my library with audible-cli it also showed these two titles and their ASINs. It may help @thefearlessleader as well. The way I saw that this was the case is by searching for the title and it showed as "in library"  but it was missing the usual download button and only gave the remove option.

**Comment by rmcrackan** [comment]
@alex-ries Note that when you 'remove' a book in Libation, it goes into your Trash Bin and does not get fully deleted. To see these books, go to Settings > Trash Bin

**Comment by thefearlessleader** [comment]
@rmcrackan wrote:

> I can verify that for US accounts, that title is not part of plus/AYCL which should result in a license denial if you don't actually own it. If you can play it in the app, then you must own it. And if you own it, you shouldn't be getting this license denial. This is the kind of oddity I would expect to resolve itself in a day or 2. However, you opened this ticket 2 days ago already. So basically: I'm stumped

Sorry, but I got hung up on other priorities...

I researched this further and found out: On my iPhone looking at the Audible app, this title (my test case out of others) showed in my library. THEN I went to look further in the browser and... it no longer is in my library. Thus, (a) the Audible app doesn't refresh a title's status, and (b) I checked several others of the "failing" titles and in the browser they now show "not in my library" or a couple "no longer available".

So, call it "my error" in not refreshing my library view or whatever, but it seems that all of those error titles must have been part of "plus" or something. Sorry 'bout believing my Audible app for saying it was in my library when it turns out it wasn't. SIGH!
